layers/best_practices/bp_pipeline.cpp

```diff
@@ -114,9 +114,10 @@ bool BestPractices::PreCallValidateCreateGraphicsPipelines(VkDevice device, VkPi
     for (uint32_t i = 0; i < createInfoCount; i++) {
         const Location create_info_loc = error_obj.location.dot(Field::pCreateInfos, i);
         const auto& create_info = pCreateInfos[i];
-        const auto& pipeline = *pipeline_states[i].get();
+        const auto pipeline = pipeline_states[i].get();
+        ASSERT_AND_CONTINUE(pipeline);
```

Comment:
```suggestion
        const auto *pipeline_ptr = pipeline_states[i].get();
        ASSERT_AND_CONTINUE(pipeline_ptr);
        const auto &pipeline = *pipeline_ptr;
```
So that the code preserves the usage of a const ref, I think it was better this way

Reply:
honestly in the future if someone works on BP again, there is a 100% chance that Pipeline becomes a large mess like it is in core check.. I'm just going to call into another function, because it isolates a single `const VkGraphicsPipelineCreateInfo* pCreateInfos` like we normally do

Reply:
made separate commit